src/utils/inner_instructions.rs

```diff
@@ -0,0 +1,57 @@
+use solana_sdk::{
+    inner_instruction::{InnerInstruction, InnerInstructionsList},
+    instruction::{CompiledInstruction, TRANSACTION_LEVEL_STACK_HEIGHT},
+    transaction_context::TransactionContext,
+};
+
+/// Pulled verbatim from `solana-svm` crate, `transaction_processor.rs`
+pub fn inner_instructions_list_from_instruction_trace(
+    transaction_context: &TransactionContext,
+) -> InnerInstructionsList {
+    debug_assert!(transaction_context
+        .get_instruction_context_at_index_in_trace(0)
+        .map(|instruction_context| instruction_context.get_stack_height()
+            == TRANSACTION_LEVEL_STACK_HEIGHT)
+        .unwrap_or(true));
+    let mut outer_instructions = Vec::new();
+    for index_in_trace in 0..transaction_context.get_instruction_trace_length() {
```

Comment:
Maybe a map following by collect is more performant, avoid new allocation each time

Reply:
Tricky to use map here since it's conditionally branching on the stack height and either pushing a new empty vec element or pushing a "sub-element" onto the last vec element.

Reply:
I could use `fold`, but since it's conditionally either pushing a new empty vec element onto the outer vec, or pushing a new "sub-element" onto the last outer vec element, it's tough to use `map` here.